client/lib/purchases/index.ts

```diff
@@ -327,16 +344,16 @@ function isObject( value ) {
  * @param {object} purchase - the purchase with which we are concerned
  * @returns {boolean}  True if the provided purchase is monthly, or false if not
  */
-function isMonthlyPurchase( purchase ) {
+export function isMonthlyPurchase( purchase: Purchase ): boolean {
 	const plan = getPlan( purchase.productSlug );
-	if ( isObject( plan ) ) {
+	if ( plan ) {
```

Comment:
`getPlan()` can only return an object or undefined, so here we really just need to know if this is truthy. That saves the need for type juggling in `isObject()`.